lib/BIGSdb/Plugins/PhyloViz.pm

```diff
@@ -0,0 +1,415 @@
+	my $query = "SELECT isolates.id, isolates.isolate FROM isolates WHERE isolates.id IN (";
+	$query .= join(",", @{$options->{'selected_ids'}});
+	$query .= ") ORDER BY isolates.id ASC";
+	my $data =  $self->{'datastore'}->run_query($query, undef, {'fetch' => 'all_arrayref'});
+
+	say q(<fieldset style="float:left"><legend>Isolates</legend>);
+	say q(<div style="float:left">);
+
+	my @ids;
+	my $labels = {};
+	foreach (@$data) {
+	    my ( $id, $isolate) = @$_;
+	    push @ids, $id;
+	    $labels->{$id} = "$id) $isolate";
+	}
+
+	say $self->popup_menu(
+	    -name     => 'isolate_id',
+	    -id       => 'isolate_id',
+	    -values   => \@ids,
+	    -labels   => $labels,
+	    -size     => $options->{'size'} // 8,
+	    -multiple => 'true',
+	    -default  => $options->{'selected_ids'},
+	    -required => $options->{'isolate_paste_list'} ? undef : 'required'
+	    );
+	say q(</div>);
+	say q(<div style="text-align:center"><input type="button" onclick='listbox_selectall("isolate_id",true)' )
+	    . q(value="All" style="margin-top:1em" class="smallbutton" />)
+	    . q(<input type="button" onclick='listbox_selectall("isolate_id",false)' value="None" )
+	    . q(style="margin-top:1em" class="smallbutton" />);
+	say q(</fieldset>);
+    }
+    return;
+}
+
+sub print_isolates_fieldset {
+        my ($self, $default_select) = @_;
+	my $q             = $self->{'cgi'};
+	my $set_id        = $self->get_set_id;
+	my $schemes       = $self->{'datastore'}->get_scheme_list( { set_id => $set_id } );
+	my $loci          = $self->{'datastore'}->get_loci_in_no_scheme( { set_id => $set_id } );
+	my $metadata_list = $self->{'datastore'}->get_set_metadata($set_id);
+	my $fields        = $self->{'xmlHandler'}->get_field_list($metadata_list);
+	my @display_fields;
+	my ( @js, @js2, @isolate_js, @isolate_js2 );
+
+	foreach my $field (@$fields) {
+		push @display_fields, $field;
+		push @display_fields, 'aliases' if $field eq $self->{'system'}->{'labelfield'};
+	}
+	push @isolate_js,  @js;
+	push @isolate_js2, @js2;
+	foreach my $field (@display_fields) {
+		( my $id = "f_$field" ) =~ tr/:/_/;
+		push @js,          qq(\$("#$id").prop("checked",true));
+		push @js2,         qq(\$("#$id").prop("checked",false));
+		push @isolate_js,  qq(\$("#$id").prop("checked",true));
+		push @isolate_js2, qq(\$("#$id").prop("checked",false));
+	}
+	say q(<fieldset style="float:left"><legend>Isolate fields</legend>);
+	$self->_print_fields(
+		{
+			fields         => \@display_fields,
+			prefix         => 'f',
+			num_columns    => 3,
+			labels         => {},
+			default_select => $default_select
+		}
+	);
+	$self->_print_all_none_buttons( \@isolate_js, \@isolate_js2, 'smallbutton' );
+	say q(</fieldset>);
+	return;
+}
+
+sub upload_data_to_phyloviz {
+
+    my $self = shift;
+    my $args = shift;
+
+    my $uuid = 0;
+    my $msg = "No message";
+    my ($data_set) = ($args->{'profile'} =~ /.+\/([^\/]+)\.txt/ );
+    my $user = $self->{'config'}->{'phyloviz_user'};
```

Comment:
OK - what I meant was, are you going to include the code for the upload script, as without this the plugin won't work?  Can we just register with Phyloviz to get the username and password?  Is this just a standard user account or do we need to contact the Phyloviz admins as we are running a service?